nixos/tests/rocm-opencl.nix

```diff
@@ -0,0 +1,17 @@
+import ./make-test-hardware.nix ({ lib, pkgs, ... } : {
```

Comment:
If we decide to move tests to individual package affected I would rather have a function like `makeImpureTest` that can be imported from nixpkgs and used outside of nixpkgs easier.